pkg/kubelet/cm/dra/claiminfo.go

```diff
@@ -21,78 +21,115 @@ import (
 	"sync"
 
 	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/kubernetes/pkg/kubelet/cm/dra/state"
 	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
 )
 
 // claimInfo holds information required
 // to prepare and unprepare a resource claim.
 type claimInfo struct {
-	sync.RWMutex
-
-	// name of the DRA driver
-	driverName string
-
-	// claimUID is an UID of the resource claim
-	claimUID types.UID
-
-	// claimName is a name of the resource claim
-	claimName string
-
-	// namespace is a claim namespace
-	namespace string
-
-	// podUIDs is a set of pod UIDs that reference a resource
-	podUIDs sets.Set[string]
-
-	// cdiDevices is a list of CDI devices returned by the
-	// GRPC API call NodePrepareResource
-	cdiDevices []string
-
+	state.ClaimInfoState
 	// annotations is a list of container annotations associated with
 	// a prepared resource
 	annotations []kubecontainer.Annotation
 }
 
-func (res *claimInfo) addPodReference(podUID types.UID) {
-	res.Lock()
-	defer res.Unlock()
-
-	res.podUIDs.Insert(string(podUID))
+// claimInfoCache is a cache of processed resource claims keyed by namespace + claim name.
+type claimInfoCache struct {
+	sync.RWMutex
+	state     state.CheckpointState
+	claimInfo map[string]*claimInfo
 }
 
-func (res *claimInfo) deletePodReference(podUID types.UID) {
-	res.Lock()
-	defer res.Unlock()
+func stateToCache(curState state.ClaimInfoStateList) (map[string]*claimInfo, error) {
```

Comment:
I don't think we need a separate function for this. I would just pull it directly into the `newClaimInfoCache()` function.

Either that or make it a `SyncFromCheckpoint()` that hangs off of a `claimInfoCache` object to match the `SyncToCheckpoint()` method I suggested below.

Then in `newClaimInfoCache()` you could call:
```
	stateImpl, err := state.NewCheckpointState(stateDir, checkpointName)
	if err != nil {
		return nil, fmt.Errorf("Could not initialize checkpoint manager, please drain node and remove dra state fil, err: %+v", err)
	}

	cache := &claimInfoCache{
		state:     stateImpl,
	}

	err = cache.SyncFromCheckpoint()
	if err != nil {
		return nil, err
	}

	return cache
```